core/tests/performance_e2e/src/metrics.cpp

```diff
@@ -0,0 +1,279 @@
+                    else
+                    {
+                        std::string deviceId(deviceId_property);
+                        METRICS_PER_DEVICE& per_device = (*module->per_device_metrics)[deviceId];
+                        per_device.messages_received++;
+                        per_device.seqence_number++;
+
+                        Counter sequence_number(std::stoll(seq_num_property));
+                        if (sequence_number != per_device.seqence_number)
+                        {
+                            per_device.out_of_sequence_messages++;
+                            if (sequence_number > per_device.seqence_number)
+                            {
+                                per_device.messages_lost += (sequence_number - per_device.seqence_number);
+                            }
+                            per_device.seqence_number = sequence_number;
+                        }
+                    }
+                }
+                catch (std::exception & e)
+                {
+                    LogError("non-conforming message: exception caught: %s", e.what());
+                    module->non_conforming_messages++;
+                }
+            }
+
+            ConstMap_Destroy(message_properties);
+        }
+    }
+}
+
+static void MetricsModule_Destroy(MODULE_HANDLE moduleHandle)
+{
+    if (moduleHandle == NULL)
+    {
+        LogError("Destroying a NULL module");
+    }
+    else
+    {
+        METRICS_MODULE_HANDLE * module = (METRICS_MODULE_HANDLE *)moduleHandle;
+        if (module->started)
+        {
+            HrTime destroy_time = std::chrono::time_point_cast<MicroSeconds>(HrClock::now());
+            MicroSeconds duration = destroy_time - module->start_time;
+            std::cout
+                << "Module Metrics:" << std::endl
+                << "---------------" << std::endl
+                << "Duration: " << duration.count() / 1000 << std::endl
+                << "Messages received: " << module->all_messages_received << std::endl
+                << "Non-Conforming Messages: " << module->non_conforming_messages << std::endl
+                << "Message Latency (average microseconds): " << module->latency.getMean().count() << std::endl
+                << "Message Latency (max microseconds): " << module->latency.max.count() << std::endl
+                << "Devices Discovered: " << module->per_device_metrics->size() << std::endl;
+            for (PerDeviceMap::iterator d = module->per_device_metrics->begin();
+                d != module->per_device_metrics->end();
+                d++)
+            {
+                std::cout
+                    << "Device: " << (*d).first << std::endl
+                    << "Message count: " << (*d).second.messages_received << std::endl
+                    << "Out of Sequence Count: " << (*d).second.out_of_sequence_messages << std::endl
+                    << "Messages Lost: " << (*d).second.messages_lost << std::endl;
```

Comment:
I think I'm going to keep it printing to the screen, because the next step is to add it as a test that runs as part of the build.